frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,156 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
+                            return G_reader_settings:readSetting("screensaver_type") == "image_file"
+                        end,
+                        keep_menu_open = true,
+                        callback = function()
+                            Screensaver:chooseFile()
+                        end,
+                    },
+                    {
+                        text = _("Select random image folder"),
+                        enabled_func = function()
+                            return G_reader_settings:readSetting("screensaver_type") == "random_image"
+                        end,
+                        keep_menu_open = true,
+                        callback = function()
+                            Screensaver:chooseFolder()
+                        end,
+                    },
+                    {
+                        text = _("Select document cover"),
+                        enabled_func = function()
+                            return G_reader_settings:readSetting("screensaver_type") == "document_cover"
+                        end,
+                        keep_menu_open = true,
+                        callback = function()
+                            Screensaver:chooseFile(true)
+                        end,
+                    },
+                },
             },
+        },
+    },
+    {
+        text = _("Sleep screen message"),
+        sub_item_table = {
             {
-                text = _("Document cover"),
-                keep_menu_open = true,
+                text = _("Add custom message to sleep screen"),
+                checked_func = function()
+                    return G_reader_settings:isTrue("screensaver_show_message")
+                end,
                 callback = function()
-                    Screensaver:chooseFile(true)
+                    G_reader_settings:toggle("screensaver_show_message")
                 end,
+                separator = true,
             },
             {
-                text = _("Screensaver message"),
-                keep_menu_open = true,
+            text = _("Edit sleep screen message"),
+            enabled_func = function()
+                return G_reader_settings:isTrue("screensaver_show_message")
+            end,
+            keep_menu_open = true,
                 callback = function()
                     Screensaver:setMessage()
                 end,
             },
             {
-                text = _("Covers and images settings"),
+                text = _("Background fill"),
+                help_text = _([[This option will only become available, if you have selected 'Put screen to sleep in current state'
+                    as screensaver and have 'Sleep screen message' on.]]),
+               enabled_func = function()
+                    return G_reader_settings:readSetting("screensaver_type") == "disable" and G_reader_settings:isTrue("screensaver_show_message")
+                end,
```

Comment:
Genuine experience with this:
I was testing things and I didn't manage to get this menu enabled. Until I read that help_text.
I was naively expecting that background mentionned here is the background of the InfoMessage (white text on black, or black text on a white background color). But it is the background of the full screen, that we may indeed want it fill instead of "Leave screen as-is". But then, we're not letting "screen as-is" :)
I would somehow expect setting the full screen background to white or black to be inside the Wallpaper>: setting a fully unicolor wallpaper.
But may be it's technically more complicated - so, really not caring, just mentionning my first time experience.

Reply:
Indentation issue here too. Better checks elsewhere too.

Reply:
I suspect this problem (the logic, not the indentation) was already there?

Reply:
May be. But the wording (and may be the absence of any enabled_func) made it less surprising:
![image](https://github.com/koreader/koreader/assets/24273478/1a4df86b-cf0d-4e6f-baae-716773c87a5d)